### PR TITLE
first draft for MASVS 6.7 iOS

### PR DESCRIPTION
**Pull request by sushi2k**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to Mitre/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectivnes of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR coveres issue #927.

### CODE REVIEW

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 1]
```diff
@@ -127,15 +127,41 @@ If it's possible to load local files via a WebView, the app might be vulnerable
 
 It should therefore be verified if a user can change the filename or path from which the file is loaded, and they shouldn't be able to edit the loaded file.
 
+
+
+### Determining Whether Native Methods Are Exposed Through WebViews
+
+#### Overview
+
+Starting from iOS version 7.0, Apple also introduced APIs that allow communication between the JavaScript runtime in the WebView and the native Swift or Objective-C app. If these APIs are used carelessly, important functionality might be exposed to attackers who manage to inject malicious script into the WebView (e.g., through a successful cross-site scripting attack).
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: Can you remove the word "also"

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 2]
```diff
@@ -127,15 +127,41 @@ If it's possible to load local files via a WebView, the app might be vulnerable
...
+
+#### Static Analysis
+
+Both `UIWebView` and `WKWebView` provide a means of communication between the WebView and the native app. Any important data or native functionality exposed to the WebView JavaScript engine would also be accessible to rogue JavaScript running in the WebView.
+
+Since iOS 7, the JavaScriptCore framework provides an Objective-C wrapper to the WebKit JavaScript engine. This makes it possible to execute JavaScript from Swift and Objective-C, as well as making Objective-C and Swift objects accessible from the JavaScript runtime. A JavaScript execution environment is represented by a `JSContext` object. Look out for code that maps native objects to the `JSContext` associated with a WebView. In Objective-C, the `JSContext` associated with a `UIWebView` is obtained as follows:
+
+``objc
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: Can you please use triple backticks (and ```obj-c)

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 3]
```diff
@@ -127,15 +127,41 @@ If it's possible to load local files via a WebView, the app might be vulnerable
...
+[webView valueForKeyPath:@"documentView.webView.mainFrame.javaScriptContext"]
+``
+
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: Can you please provide some bridging/guiding text here before you jump to the enumeration: what is actually wrong with what you are going to describe?

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 4]
```diff
@@ -127,15 +127,41 @@ If it's possible to load local files via a WebView, the app might be vulnerable
...
+- Objective-C blocks. When an Objective-C block is assigned to an identifier in a JSContext, JavaScriptCore automatically wraps the block in a JavaScript function;
+- JSExport protocol: Properties, instance methods, and class methods declared in a JSExport-inherited protocol are mapped to JavaScript objects that are available to all JavaScript code. Modifications of objects that are in the JavaScript environment are reflected in the native environment.
+
+Note that only class members defined in the `JSExport` protocol are made accessible to JavaScript code.
+
+#### Dynamic Analysis
+
+Dynamic analysis of the app can show you which HTML or JavaScript files are loaded and which vulnerabilities are present. The procedure for exploiting the vulnerability starts with producing a JavaScript payload and injecting it into the file that the app is requesting. The injection can be accomplished via a MITM attack. See an example for a vulnerable iOS in [#THIEL] page 156 following.
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: Now we need a book for this. Is it possible to enumerate on the attack so it becomes clear what a pentester has to do?

File: Document/0x06h-Testing-Platform-Interaction.md [hunk 5]
```diff
@@ -127,15 +127,41 @@ If it's possible to load local files via a WebView, the app might be vulnerable
...
-
-Besides potential script injection, there's another fundamental WebViews security issue: the WebKit libraries packaged with iOS don't get updated out-of-band like the Safari web browser. Therefore, newly discovered WebKit vulnerabilities remain exploitable until the next full iOS update [#THIEL].
+iOS WebViews support JavaScript execution by default, so script injection and cross-site scripting attacks can affect them. Besides potential script injection, there's another fundamental WebViews security issue: the WebKit libraries packaged with iOS don't get updated out-of-band like the Safari web browser. Therefore, newly discovered WebKit vulnerabilities remain exploitable until the next full iOS update [#THIEL].
```
Comment (commjoen) [inline Document/0x06h-Testing-Platform-Interaction.md]: Can you show where we can find them online?

**Comment by commjoen** [comment]
Finally, don't forget to add a reference in the attack surface section :)

File: Document/0x06h-Testing-Platform-Interaction.md (hunk 1 above)
Comment (sushi2k) [reply Document/0x06h-Testing-Platform-Interaction.md]: done

File: Document/0x06h-Testing-Platform-Interaction.md (hunk 2 above)
Comment (sushi2k) [reply Document/0x06h-Testing-Platform-Interaction.md]: done

File: Document/0x06h-Testing-Platform-Interaction.md (hunk 3 above)
Comment (sushi2k) [reply Document/0x06h-Testing-Platform-Interaction.md]: Updated the text before

**Pull request merged**